align/schema/checker.py

```diff
@@ -158,7 +158,11 @@ def append(self, formula, label=None):
                 labels=self._solver.unsat_core())
 
     def label(self, object):
-        return z3.Bool(str(id(object)))
+        # Z3 throws 'index out of bounds' error
```

Comment:
Is this 9 decimal digits or more than 32 bits? Do we know which?

Reply:
I was unable to find any documentation on this. I arrived at the % 10**9 pretty much by trial and error.

We can try hashing with hashlib and extracting the hexdigest instead of the inbuilt hash if we do see hash conflicts. Should give us a bigger range. This was the simplest and quickest is all.

Reply:
Looking at the code a little it seems to switch on whether the argument is an integer or a string. If it is an integer it probably depends on whether it can be represented in C. This can probably be fixed by making it  a string if you want.